Added replan script on buttonclick

nalvoj opened the pull request.

nalvoj committed:
Added Search Bar feature

nalvoj committed:
Revert "Added Search Bar feature"

This reverts commit 754956e173d8792da4ea22610b710817cbe49741.

nalvoj committed:
Beginning SearchBar feature

updated APIs, added SearchBar in html and css, granted functionality of SearchBar in map.js

nalvoj committed:
Done SearchBar

nalvoj committed:
Added replan button, made SearchBar collapsible

nalvoj committed:
Added Replan function on button

aleskucera reviewed (changes requested):
1. The replanning doesn't work and I don't even see where might be a problem. Every input gets error message: `Could not found new route`. Make sure that the feature works every time or prints the reason why it didn't work.

2. I didn't know what to install. Please update the dependencies of in `pyproject.toml` and the `uv.lock`.

3. There is no comment in the PR. No explanation of the feature at all.

vlk-jan committed:
update module loading and gitignore

nalvoj committed:
Replan function modification

Replan function recognizes multiple outputs (such as No need to replan). When path not found - RRT* tries it few more times. Added overlay when replanning
Updated readme and pyproject to stand to those modifications

vlk-jan committed:
bug fixes, code cleanup, fix imports

vlk-jan committed:
fix grid bug in rrt*

vlk-jan committed:
code cleanup and optimization

vlk-jan commented:
Problem with RRT* should be mostly resolved. May still fail, but those should only be when no path is possible or in typical anti-RRT situations.

vlk-jan committed:
update dependencies

vlk-jan commented:
`pyproject.toml` and `uv.lock` updated.

vlk-jan committed:
multithreaded rrt replanning

vlk-jan commented:
Added multithreading solving #7.

vlk-jan committed:
add inflation of obstacles

vlk-jan commented:
Added inflation of obstacles solving #6.
